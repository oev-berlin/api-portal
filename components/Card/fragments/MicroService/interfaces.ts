interface MicroServiceProps {
    serviceName: string,
    isMicroservice: boolean,
}

export type { MicroServiceProps };
