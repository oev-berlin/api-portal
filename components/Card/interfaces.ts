import {projectData} from "../../utils/interfaces";

interface MicroServiceProps{
    serviceName: string
}

interface CardProps{
    projectData: projectData
}

export {MicroServiceProps, CardProps}