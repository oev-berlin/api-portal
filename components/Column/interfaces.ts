import {projectData} from "../../utils/interfaces";

interface columnData {
    name: string,
    projects: projectData[],
}

export {columnData};