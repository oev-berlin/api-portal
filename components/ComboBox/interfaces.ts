export interface ComboBoxProps{
    searchOptions: string[],
}
