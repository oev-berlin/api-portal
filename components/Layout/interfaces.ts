import React from 'react';

export interface LayoutProps {
    searchOptions: string[],
    children: React.ReactNode,
}
