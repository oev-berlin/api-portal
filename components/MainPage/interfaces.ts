type docType = "backend" | "microservice";

interface projectData {
    id: string,
    name: string,
    description: string,
    swaggerInformation: object,
    docsType: docType,
    microservices: string[],
    externalServices: string[]
}

interface columnData {
    name: string,
    projects: JSON[]
}

export {projectData, columnData};

