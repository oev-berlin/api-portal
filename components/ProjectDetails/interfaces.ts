interface ProjectDetailsProps {
    name: string|undefined,
    description?: string|undefined,
}

export type { ProjectDetailsProps };
