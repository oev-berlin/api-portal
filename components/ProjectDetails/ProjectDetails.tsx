import React from 'react';
import { StyledH1, Description, Text } from './styles';
import { projectData } from '../../utils/interfaces';

export const ProjectDetails = ({ project }: { project: projectData }) => (
  <>
    <StyledH1> {project.name} </StyledH1>
    <Description>Description</Description>
    <Text>{project.description}</Text>
  </>
);
