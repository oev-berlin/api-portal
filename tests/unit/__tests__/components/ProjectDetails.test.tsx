import { render, cleanup } from '@testing-library/react';
import { toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { projectData } from '../../../../utils/interfaces';
// eslint-disable-next-line import/extensions
import 'jest-styled-components';
import { ProjectDetails } from '../../../../components/ProjectDetails';
import exampleProject from '../../../../public/apis_docs/example_1.json';

expect.extend(toHaveNoViolations);

const testProject1: projectData = exampleProject;
const testProject2: projectData = { ...exampleProject, name: '' };
const testProject3: projectData = { ...exampleProject, description: '' };

const setupComponent = ({ project }: { project: projectData }) => render(
  <ProjectDetails project={project} />,
);

describe('ProjectDetails', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('ProjectDetails component (project name and project description are not empty) should render without crashing', async () => {
      const screen = setupComponent({
        project: testProject1,
      });
      expect(screen.container).toBeDefined();
    });
    it('ProjectDetails component (project name is empty) should render without crashing', async () => {
      const screen = setupComponent({
        project: testProject2,
      });
      expect(screen.container).toBeDefined();
    });
    it('ProjectDetails component (project description is empty) should render without crashing', async () => {
      const screen = setupComponent({
        project: testProject3,
      });
      expect(screen.container).toBeDefined();
    });

    it('ProjectDetails component should render correctly', async () => {
      const screen = setupComponent({ project: testProject1 });
      await screen
        .findByText(/project a/i)
        .then((element) => expect(element).toBeInTheDocument());
      await screen
        .findByText(/Description/)
        .then((element) => expect(element).toBeInTheDocument());
      await screen
        .findByText(/description of the project/)
        .then((element) => expect(element).toBeInTheDocument());
    });
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot (project name and project description are not empty)', () => {
      const { container } = setupComponent({ project: testProject1 });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (project name is empty)', () => {
      const { container } = setupComponent({ project: testProject2 });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (project description is empty)', () => {
      const { container } = setupComponent({ project: testProject3 });
      expect(container).toMatchSnapshot();
    });
  });
});
