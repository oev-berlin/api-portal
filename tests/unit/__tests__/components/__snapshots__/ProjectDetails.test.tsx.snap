// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetails Snapshots Tests should match a basic snapshot (project description is empty) 1`] = `
.c0 {
  text-align: left;
  margin-bottom: 1rem;
  margin-left: 1rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 0.5rem;
}

.c1 {
  text-align: left;
  margin-left: 2rem;
}

.c2 {
  margin-left: 3rem;
  margin-right: 3rem;
  text-align: justify;
}

<div>
  <h1
    class="c0"
  >
     
    Project A
     
  </h1>
  <h2
    class="c1"
  >
    Description
  </h2>
  <div
    class="c2"
  />
</div>
`;

exports[`ProjectDetails Snapshots Tests should match a basic snapshot (project name and project description are not empty) 1`] = `
.c0 {
  text-align: left;
  margin-bottom: 1rem;
  margin-left: 1rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 0.5rem;
}

.c1 {
  text-align: left;
  margin-left: 2rem;
}

.c2 {
  margin-left: 3rem;
  margin-right: 3rem;
  text-align: justify;
}

<div>
  <h1
    class="c0"
  >
     
    Project A
     
  </h1>
  <h2
    class="c1"
  >
    Description
  </h2>
  <div
    class="c2"
  >
    description of the project
  </div>
</div>
`;

exports[`ProjectDetails Snapshots Tests should match a basic snapshot (project name is empty) 1`] = `
.c0 {
  text-align: left;
  margin-bottom: 1rem;
  margin-left: 1rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 0.5rem;
}

.c1 {
  text-align: left;
  margin-left: 2rem;
}

.c2 {
  margin-left: 3rem;
  margin-right: 3rem;
  text-align: justify;
}

<div>
  <h1
    class="c0"
  >
     
    
     
  </h1>
  <h2
    class="c1"
  >
    Description
  </h2>
  <div
    class="c2"
  >
    description of the project
  </div>
</div>
`;
